contracts/ClaimSettlementBase.sol

```diff
@@ -87,7 +87,7 @@ abstract contract ClaimSettlementBase is Module {
         keccak256("TransferNFTToCaller(address token,uint256 tokenId)");
 
     modifier onlyAvatar() {
-        require(msg.sender == avatar, "caller is not the right avatar");
+        require(_msgSender() == avatar, "caller is not the right avatar");
```

Comment:
Cool. Yea here is an example of [use-case ](https://ethereum.stackexchange.com/questions/110522/what-does-msgsender-this-code-do-in-the-opensea-repositorys-example-erc721)